### DISCUSSION TITLE
is there a way to drop a table?

### DISCUSSION THREAD

**Discussion by dimitri-b** [erezsh/Preql#32] [Q&A] [answered]
I looked through documentation - I can delete rows from a table via `table_name delete[foo == "bar"]`, but I couldn't find any reference to dropping a table. Is it supported?

**Comment by erezsh** [comment, marked as answer]
Yes, you can call `remove_table()`

Like,

```go
remove_table(my_table)
```

**Reply by dimitri-b** [reply]
I tried that and got some mixed results.
I'm connected to SQLite, calling `remove_table(foo)` - get no error but `tables()` still lists the table. Querying it returs `=0` - it appears to be empty.
I also tried adding `commit()` - but I get  `cannot commit - no transaction is active` error.
I tried both Jupyter and Terminal.

**Reply by erezsh** [reply]
Yes, you're right. It is a bug, sorry!

You can solve it for now by declaring it yourself:

```go
func remove_table(table_name) {
    "Remove table from database (drop table)"
    force_eval(SQL(nulltype, "DROP TABLE $table_name"))
}
```

I fixed it now, so it will work in the next release.

**Reply by erezsh** [reply]
Btw, Jupyter and Terminal should always return the same result. If they do not, it is a bug on its own!
